### ISSUE TITLE
Allow to name #[TestWith] and #[TestWithJson] data sets

### ISSUE THREAD

**Issue by vudaltsov** [sebastianbergmann/phpunit#5713]
If I'm not mistaken, right now there is no way to name inline data sets. I propose to add an optional `$name` parameter to attributes:

```php
final class DataTest extends TestCase
{
    #[TestWith([0, 0, 0], 'first data set')]
    #[TestWithJson('[0, 1, 1]', 'second data set')]
    public function testAdd(int $a, int $b, int $expected): void
    {
        $this->assertSame($expected, $a + $b);
    }
}
```

**Comment by sebastianbergmann** [comment]
Was this not implemented in #4964 for PHPUnit 11.0?

**Comment by vudaltsov** [comment]
Cool, thanks! Haven't updated yet.
